adapter-tcp/src/main/java/io/nosqlbench/adapter/tcpserver/TcpServerAdapterSpace.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2022 nosqlbench
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.nosqlbench.adapter.tcpserver;
+
+import io.nosqlbench.api.config.standard.ConfigModel;
+import io.nosqlbench.api.config.standard.NBConfigModel;
+import io.nosqlbench.api.config.standard.NBConfiguration;
+import io.nosqlbench.api.config.standard.Param;
+import io.nosqlbench.api.engine.activityimpl.ActivityDef;
+import io.nosqlbench.api.engine.util.SSLKsFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import java.net.ServerSocket;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import javax.net.ServerSocketFactory;
+import javax.net.ssl.SSLServerSocketFactory;
+
+public class TcpServerAdapterSpace implements AutoCloseable{
+    @Override
+    public void close() throws Exception {
+        logger.info("TcpServerAdapterSpace is waiting for message queue to empty");
+        while(this.queue != null && !this.queue.isEmpty())
+        {
+            try {
+                Thread.sleep(10);
+            }catch (InterruptedException e) {
+            }
+        }
+        logger.info("TcpServerAdapterSpace is being closed");
+        for (Shutdown toClose : managedShutdown) {
+            toClose.shutdown();
+        }
+    }
+
+    private final static Logger logger = LogManager.getLogger(TcpServerAdapterSpace.class);
+    private final NBConfiguration config;
+    Writer writer;
+    private LinkedBlockingQueue<String> queue;
+    private ServerSocket listenerSocket;
+    private final List<Shutdown> managedShutdown = new ArrayList<>();
+    private int capacity=10;
+    private ServerSocketFactory socketFactory;
```

Comment:
This can be removed the instance variable level and used in the method where it used.